tt_metal/programming_examples/contributed/vecadd/vecadd.cpp

```diff
@@ -0,0 +1,215 @@
+{
+    CircularBufferConfig cb_src0_config = CircularBufferConfig(
+        size,
+        {{
+            cb,
+            format
+    }})
+    .set_page_size(cb, page_size);
+    return CreateCircularBuffer(program, core, cb_src0_config);
+}
+
+// Circular buffers are Tenstorrent's way of communicating between the data movement and the compute kernels.
+// kernels queue tiles into the circular buffer and takes them when they are ready. The circular buffer is
+// backed by SRAM. There can be multiple circular buffers on a single Tensix core.
+// @param program: The program to create the circular buffer on.
+// @param core: The core to create the circular buffer on.
+// @param cb: Which circular buffer to create (c_in0, c_in1, c_out0, c_out1, etc..). This is just an ID
+// @param n_tiles: The number of tiles the circular buffer can hold.
+CBHandle MakeCircularBufferBFP16(Program& program, const CoreSpec& core, tt::CB cb, uint32_t n_tiles)
+{
+    constexpr uint32_t tile_size = 2 * (32 * 32);
+    return MakeCircularBuffer(program, core, cb, n_tiles * tile_size, tile_size, tt::DataFormat::Float16_b);
+}
+
+std::string next_arg(int& i, int argc, char **argv)
+{
+    if(i + 1 >= argc) {
+        std::cerr << "Expected argument after " << argv[i] << std::endl;
+        exit(1);
+    }
+    return argv[++i];
+}
+
+void help(std::string_view program_name)
+{
+    std::cout << "Usage: " << program_name << " [options]\n";
+    std::cout << "This program demonstrates how to add two vectors using tt-Metalium.\n";
+    std::cout << "\n";
+    std::cout << "Options:\n";
+    std::cout << "  --device, -d <device_id>  Specify the device to run the program on. Default is 0.\n";
+    std::cout << "  --seed, -s <seed>         Specify the seed for the random number generator. Default is random.\n";
+    exit(0);
+}
+
+int main(int argc, char **argv)
+{
+    int seed = std::random_device{}();
+    int device_id = 0;
+
+    // Quick and dirty argument parsing.
+    for(int i = 1; i < argc; i++) {
+        std::string_view arg = argv[i];
+        if(arg == "--device" || arg == "-d") {
+            device_id = std::stoi(next_arg(i, argc, argv));
+        }
+        else if(arg == "--seed" || arg == "-s") {
+            seed = std::stoi(next_arg(i, argc, argv));
+        }
+        else if(arg == "--help" || arg == "-h") {
+            help(argv[0]);
+            return 0;
+        }
+        else {
+            std::cout << "Unknown argument: " << arg << std::endl;
+            help(argv[0]);
+        }
+    }
+
+    Device *device = CreateDevice(device_id);
+
+    Program program = CreateProgram();
+    // This example program will only use 1 Tensix core. So we set the core to {0, 0}.
+    constexpr CoreCoord core = {0, 0};
+
+    CommandQueue& cq = device->command_queue();
+    const uint32_t n_tiles = 64;
+    const uint32_t tile_size = 32 * 32;
```

Comment:
Similar comment to above. Should use the defined values `TILE_HEIGHT * TILE_WIDTH` or `TILE_HW`